util/cache/redis.go

```diff
@@ -26,6 +30,42 @@ type redisCache struct {
 	expiration time.Duration
 	client     *redis.Client
 	cache      *rediscache.Cache
+	compress   bool
+}
+
+func (r *redisCache) marshal(obj interface{}) ([]byte, error) {
+	buf := bytes.NewBuffer([]byte{})
+	var w io.Writer = buf
+	if r.compress {
+		w = gzip.NewWriter(buf)
+	}
+	encoder := json.NewEncoder(w)
+
+	if err := encoder.Encode(obj); err != nil {
+		return nil, err
+	}
+	if flusher, ok := w.(interface{ Flush() error }); ok {
+		if err := flusher.Flush(); err != nil {
+			return nil, err
+		}
+	}
+	return buf.Bytes(), nil
+}
+
+func (r *redisCache) unmarshal(data []byte, obj interface{}) error {
+	buf := bytes.NewReader(data)
+	var reader io.Reader = buf
+	if r.compress {
+		if r, err := gzip.NewReader(buf); err != nil {
+			if err == gzip.ErrHeader {
+				return ErrCacheMiss
```

Comment:
Is this the mechanism to switch over to gzipped contents? Throw a cache miss, forcing a re-population of the cache?

If so, should we log the cache miss? We'd expect the cache to quickly be switched over to all gzipped contents. Logging the miss would allow us to debug if the header issue persists.

Reply:
Yes, this was an attempt to switch to compression automatically. I've realized did not think about disabling compression.  Instead of analyzing error I think it is better to just use different key.

I'm adding `.gz` postfix when compression is enabled. E.g. `app|managed-resources|guestbook|1.8.3.gz` . So we have a way to change compression mechanism and use a different "extension".

Reply:
Yep, I like that!